FEAT: Support for building Windows package.

timcera opened the pull request:
MNT: Re-rendered with conda-build 3.28.1, conda-smithy 3.30.2, and conda-forge-pinning 2024.01.03.13.20.35

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

timcera committed:
FEAT: Support for building Windows package.

MNT: Re-rendered with conda-build 3.28.1, conda-smithy 3.30.2, and conda-forge-pinning 2024.01.03.13.20.35

EricEngle-NOAA commented on .ci_support/win_64_.yaml:
```diff
@@ -0,0 +1,14 @@
+target_platform:
+- win-64
+zlib:
+- '1.2'
```
Can we add support for AEC compression?  Looks like libaec is available in conda

https://anaconda.org/conda-forge/libaec

timcera replied:
This file is created by "conda smithy rerender" and I don't know why libaec isn't included here, but others are.  

libaec is listed as a host requirement for all platforms in the recipe/meta.yaml.

My hypothesis is that libaec isn't listed in .ci_support/win_64_.yaml because I put minimum version constraint on libaec in recipe/meta.yaml, but not the other packages.

xylar replied:
I think you need to list `libaec` once without a constraint and once with a constraint in `host` for conda-smithy to pick it up and put it in the `.ci_support` files as it should.  I've just added this.

EricEngle-NOAA reviewed (changes requested):
See comment about libaec.

xylar committed:
Add libaec again without constraint for conda-smithy

xylar commented:
@conda-forge-admin, please rerender.

xylar commented on recipe/meta.yaml:
```diff
@@ -10,39 +10,40 @@ source:
       host:
         - libpng
         - zlib
+        # list libaec twice to conda-smithy doesn't get confused
+        - libaec
         - libaec >=1.0.6
```
This looks a little weird but I think this is how to make conda-smithy work right.

Commit:
MNT: Re-rendered with conda-build 3.28.2, conda-smithy 3.30.2, and conda-forge-pinning 2024.01.04.06.55.30

timcera committed:
On Windows moved to Jasper instead of OpenJPEG

OpenJPEG puts it's headers in a sub-directory
"..\include\openjeg-2.5" which "cmake" doesn't
discover on Windows.  Jasper puts it's header
file in the include directory so hope that
"cmake" finds it.

timcera committed:
Merge remote-tracking branch 'refs/remotes/origin/windows_test' into windows_test

timcera commented:
Not successful yet - let me do some research on how to get "unistd.h" on Windows (heavily favored approach) or propose edits to NCEPLIBS-g2c to make it play nicer on Windows.

timcera committed:
Patch grib2.h.in to not include the unused unistd.h

timcera committed:
Patch so that "m" lib isn't included on Windows.

timcera committed:
Wrong name of the library on Windows. Should be g2c.dll not libg2c.dll.

xylar reviewed (approved):
Wonderful, thanks @timcera
